### DISCUSSION TITLE
How can I create a folder structure with default files using command in adonis?

### DISCUSSION THREAD

**Discussion by hkmsadek** [adonisjs/core#1329] [Help] [answered]
Hello there, I am new to package development things, I would like to create something using which I can code quickly and do my code structure nicely. For example, in adonis, we can create a controller file like `adonis make:controller MyController`, I want to create some a folder with some files with a custom command. For example one folder and three `.js` file under it with some boiler plate codes. How can I do this or which documentation I should I read? Need some guideline. 

Many thanks..

**Comment by thetutlage** [comment]
Which version of the framework are you using? `@adonisjs/core` version

**Reply by hkmsadek** [reply]
Thank you. I am using 4.1 but want to learn this for 5 as well.

**Comment by McSneaky** [comment, marked as answer]
To create files and populate them you can use NodeJS FS: 
https://nodejs.org/api/fs.html

fs.promises are worth a look at: https://nodejs.org/api/fs.html#fs_fs_promises_api can use await for them

To make custom ace command follow the docs: 
https://adonisjs.com/docs/4.1/ace#_creating_commands
